lib/clerk/sdk.rb

```diff
@@ -106,5 +110,40 @@ def sms_messages
     def users
       Resources::Users.new(self)
     end
+
+    # Returns the decoded JWT payload without verifying if the signature is
+    # valid.
+    #
+    # WARNING: This will not verify whether the signature is valid. You
+    # should not use this for untrusted messages! You most likely want to use
+    # verify_token.
+    def decode_token(token)
+      JWT.decode(token, nil, false).first
+    end
+
+    # Decode the JWT and verify it's valid (verify claims, signature etc.)
+    #
+    # JWKS are cached for JWKS_CACHE_LIFETIME seconds, in order to avoid
+    # unecessary roundtrips. In order to invalidate the cache, pass
+    # `force_refresh_jwks: true`.
+    def verify_token(token, force_refresh_jwks: false)
+      unverified_claims = decode_token(token)
+
+      jwk_loader = ->(options) do
+        @cached_jwks = nil if options[:invalidate] || force_refresh_jwks
```

Comment:
❓ How does this caching work with the instance variable? Shouldn't the cache be on the class level (Ruby class or self variables) ?

Reply:
The user is supposed to instantiate once a `Clerk::SDK` object and re-use it across requests to the Clerk API.

This caching here works based on that pattern. So as long as the same SDK client is re-used, it's going to leverage this cache. It's a per-SDK client cache. I don't think there's a reason to put the cache at the class level as it would complicate things and it's generally an anti-pattern.